pkg/core/proxy/integrations/http/http.go

```diff
@@ -129,6 +128,12 @@ func (h *HTTP) parseFinalHTTP(ctx context.Context, mock *FinalHTTP, destPort uin
 			utils.LogError(h.Logger, err, "failed to read the http request body", zap.Any("metadata", utils.GetReqMeta(req)))
 			return err
 		}
+
+		reqBody, err = pkg.Decompress(h.Logger, req.Header.Get("Content-Encoding"), reqBody)
```

Comment:
Add a condition here that if value of "Content-Encoding" is not empty then only Call that function, better for readability.